internal/proxy/proxy_test.go

```diff
@@ -35,22 +37,27 @@ func TestApply(t *testing.T) {
 		wantUnchangedFiles []string
 		wantErr            bool
 	}{
-		"No options set, no environment file is created":       {},
-		"No options set, previous environment file is deleted": {prevContents: map[string]string{envConfigPath: "HTTP_PROXY=http://example.com:8080"}},
+		"No options set, no configuration files are created":       {},
+		"No options set, previous configuration files are deleted": {prevContents: map[string]string{envConfigPath: "HTTP_PROXY=http://example.com:8080", aptConfigPath: `Acquire::http::Proxy "http://example.com:8080";`}},
 		"HTTP option set":  {http: "http://example.com:8080"},
 		"Some options set": {http: "http://example.com:8080", https: "https://example.com:8080"},
-		"Some options set, environment parent directories are created": {
+		"Some options set, configuration parent directories are created": {
 			http: "http://example.com:8080", https: "https://example.com:8080", existingDirs: []string{},
 		},
-		"Some options set, environment file should not be changed": {
+		"Some options set, configuration files should not be changed": {
 			http: "http://example.com:8080", https: "https://example.com:8080",
-			prevContents: map[string]string{envConfigPath: fmt.Sprintf(`%s
+			prevContents: map[string]string{
+				envConfigPath: fmt.Sprintf(`%s
 HTTP_PROXY=http://example.com:8080
 http_proxy=http://example.com:8080
 HTTPS_PROXY=https://example.com:8080
 https_proxy=https://example.com:8080
+`, proxy.ConfHeader),
+				aptConfigPath: fmt.Sprintf(`%s
+Acquire::http::Proxy "http://example.com:8080";
+Acquire::https::Proxy "https://example.com:8080";
 `, proxy.ConfHeader)},
-			wantUnchangedFiles: []string{envConfigPath},
+			wantUnchangedFiles: []string{envConfigPath, aptConfigPath},
```

Comment:
I would still add some separate use case to deal with env file only and their refresh policies (file updated, file up to date), and same for apt file.

That way, if there are some failures only on "one side", you will be able to pinpoint exactly where the issue is coming from.

Also, a couple of additional "Mixed" use case, when one file is up to date, but not the other (and vice-versa) as a `// special` stenza.

Reply:
ah no, still doing package tests, but the initial state only contain "one side" or we only compare "one side". At least, the minimum I think would be to have inconsistent state (one already up to date, like env, and one which isn’t, like apt) and vice-versa